pubsub/rabbitmq/rabbitmq.go

```diff
@@ -401,6 +406,57 @@ func (r *rabbitMQ) prepareSubscription(channel rabbitMQChannelBroker, req pubsub
 		args[argMaxPriority] = mp
 	}
 
+	// queue type is classic by default, but we allow user to create quorum queues if desired
+	if val, ok := req.Metadata[metadataQueueType]; ok && val != "" {
+		if !queueTypeValid(val) {
+			return nil, errors.New(fmt.Sprintf("Invalid queue type %s. Valid types are %s, %s and %s", val, queueTypeClassic, queueTypeQuorum, queueTypeStream));
+		} else {
+			args[argQueueType] = val
+		}
+	} else if val == "" {
+		args[argQueueType] = queueTypeClassic
+	}
+
+	// Applying x-max-length-bytes if defined
+	if val, ok := req.Metadata[metadataMaxLenBytesKey]; ok && val != "" {
+		parsedVal, pErr := strconv.ParseUint(val, 10, 0)
+		if pErr != nil {
+			r.logger.Errorf("%s prepareSubscription error: can't parse %s value on subscription metadata for topic/queue `%s/%s`: %s", logMessagePrefix, argMaxLengthBytes, req.Topic, queueName, pErr)
+			return nil, pErr
+		}
+		args[argMaxLengthBytes] = parsedVal
+	}
+
+	//Applying x-max-length if defined, but only for quorum and classic queues
+	if val, ok := req.Metadata[metadataMaxLenKey]; ok && val != "" {
+		if args[argQueueType] == queueTypeStream {
+			errMsg := fmt.Sprintf("%s prepareSubscription error: can't apply %s to stream queues on subscription metadata for topic/queue `%s/%s`", logMessagePrefix, argMaxLength, req.Topic, queueName)
+			r.logger.Errorf(errMsg)
+			return nil, errors.New(errMsg)
+		}
+		parsedVal, pErr := strconv.ParseUint(val, 10, 0)
+		if pErr != nil {
+			r.logger.Errorf("%s prepareSubscription error: can't parse %s value on subscription metadata for topic/queue `%s/%s`: %s", logMessagePrefix, argMaxLength, req.Topic, queueName, pErr)
+			return nil, pErr
+		}
+		args[argMaxLength] = parsedVal
+	}
+
+	// Applying x-max-age if definde, but only for stream queues
+	if val, ok := req.Metadata[metadataMaxAgeKey]; ok && val != "" {
```

Comment:
If this is not defined at the request level do you want to use the global default you define in the component metadata? `r.rabbitMQMetadata.MaxAge` -- you made that value configurable but don't use it anywhere.

Reply:
You are right. My fault. As it only applies to stream queues, I'm going to change it to be defined in req metadata